File: src/za/co/adhd_developers/java/questions/Question38.java
package za.co.adhd_developers.java.questions;

import za.co.adhd_developers.java.tools.Question;
import za.co.adhd_developers.java.tools.Utils;

public class Question38 implements Question
{
    private int answer = 0;
    @Override
    public void doWork()
    {
        for (int base = 0; base < 1000000; base++)
        {
            String hold = "";
            hold += String.valueOf(base*1);
            hold += String.valueOf(base*2);
            int n = 2;

            while (hold.length() < 9)
            {
                hold += String.valueOf(base * n);
                n++;
            }

            if (Utils.isPandigital(hold, 9))
            {
                int newPossible = Integer.valueOf(hold);
                if (newPossible > this.answer)
                {
                    this.answer = newPossible;
                }
            }
        }
    }

    @Override
    public void printAnswer()
    {
        System.out.println("=================================");
        System.out.println("Question 38");
        System.out.println("The largest 1 to 9 pandigital 9-digit number that can be formed as the concatenated product of an integer with (1,2, ... , n) where n > 1, is: " + this.answer);
        System.out.println("=================================");
    }
}

File: src/za/co/adhd_developers/kotlin/questions/Question38.kt
package za.co.adhd_developers.kotlin.questions

import za.co.adhd_developers.kotlin.Question
import za.co.adhd_developers.kotlin.tools.isPandigital

class Question38 : Question {

    var answer = 0

    override fun doWork() {

        for (i in 1 until 1000000) {
            var hold = ""
            hold += i.times(1).toString()
            hold += i.times(2).toString()
            var n = 1

            while (hold.length < 9) {
                hold += i.times(n).toString()
                n++
            }

            if (hold.length != 9) {
                continue
            }

            if (isPandigital(hold)) {
                if (this.answer < hold.toInt()) {
                    this.answer = hold.toInt()
                }
            }
        }

    }

    override fun printAnswer() {
        println("=================================")
        println("Question 38")
        println("The largest 1 to 9 pandigital 9-digit number that can be formed as the concatenated product of an integer with (1,2, ... , n) where n > 1, is: " + this.answer)
        println("=================================")

    }
}